### PR TITLE
Storage/Added Owner, Group, and Permissions

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp [hunk 1]
```diff
@@ -344,6 +344,12 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     ret.IsDirectory = _detail::MetadataIncidatesIsDirectory(ret.Metadata);
     ret.EncryptionContext = Azure::Core::Http::_internal::HttpShared::GetHeaderOrEmptyString(
         response.RawResponse->GetHeaders(), _detail::EncryptionContextHeaderName);
+    ret.Owner = Azure::Core::Http::_internal::HttpShared::GetHeaderOrEmptyString(
```
[the remark this answers is not in the document]
Comment (microzchang) [reply sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp:347]: Oh, yeah! I will fix this.

File: sdk/storage/azure-storage-files-datalake/test/ut/datalake_file_client_test.cpp [hunk 2]
```diff
@@ -609,6 +609,14 @@ namespace Azure { namespace Storage { namespace Test {
     EXPECT_THROW(ReadFile(testName), std::runtime_error);
   }
 
+  TEST_F(DataLakeFileClientTest, FileDownloadOwnerGroupPermissions)
+  {
+    auto downloadResult = m_fileClient->Download().Value;
+    ASSERT_TRUE(downloadResult.Details.Owner.HasValue());
+    ASSERT_TRUE(downloadResult.Details.Group.HasValue());
```
[the remark this answers is not in the document]
Comment (microzchang) [reply sdk/storage/azure-storage-files-datalake/test/ut/datalake_file_client_test.cpp:616]: After not using `GetHeaderOrEmptyString` , I think whether it's empty-string is a server behavior?

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp]: Can you check .Net SDK? Are these fields nullable?

File: sdk/storage/azure-storage-files-datalake/test/ut/datalake_file_client_test.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/test/ut/datalake_file_client_test.cpp:616]: Will service ever return empty string for these 3 fields? If not, I think it still makes sense to reassure the values are correctly parsed from raw response.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**